tests/test_petab.py

```diff
@@ -270,7 +270,7 @@ def test_partial_override(self, condition_df_2_conditions):
                      'dynamicParameter2',
                      'obs1par1override',
                      'obs1par2cond1override',
-                     'observableParameter1_obs2',
+                     np.nan,
```

Comment:
had to adapt the expected behavior here. not sure if that is ok, or if one would expect to have non-overrides still in there, and apply handle_missing_overrides only afterwards explicitly

Reply:
Yeah, that test was from back where I thought we would allow that...

Reply:
but it is good that the test exists already that captures this case. so I don't have to write any :)

Reply:
My pleasure